hw_diag/utilities/hardware.py

```diff
@@ -188,15 +190,36 @@ def detect_ecc(diagnostics):
     # It determines which value will be available for which key
     variant = diagnostics.get('VA')
 
+    i2c_bus = ''
     try:
-        key_storage_bus = get_variant_attribute(variant, 'KEY_STORAGE_BUS')
+        # Example SWARM_KEY_URI: "ecc://i2c-1:96?slot=0"
+        swarm_key_uri = get_variant_attribute(variant, 'SWARM_KEY_URI')
+        parse_result = urlparse(swarm_key_uri)
+        pattern = 'i2c-([0-9]+)'
+        i2c_bus = re.search(pattern, parse_result.hostname).group(1)
+
     except Exception as e:
+        logging.warn("Unable to lookup SWARM_KEY_URI from hardware definitions, "
+                     "Exception message: {}".format(e))
+
+    if not i2c_bus or not i2c_bus.isnumeric():
+        try:
+            # Example KEY_STORAGE_BUS: "/dev/i2c-1"
+            key_storage_bus = get_variant_attribute(variant, 'KEY_STORAGE_BUS')
+            pattern = 'i2c-([0-9]+)'
```

Comment:
this is used multiple times. Make a small function. I think old method of splitting at "-" was also ok, we don't expect anything else than i2c-<number> to come here. But I don't have any problem with additional error checking.

Reply:
Inner utility method for parsing i2c_bus address looks better.